crates/storage/provider/src/traits/chain.rs

```diff
@@ -0,0 +1,140 @@
+///! Canonical chain state notification trait and types.
+use crate::PostState;
+use auto_impl::auto_impl;
+use core::fmt;
+use reth_primitives::{BlockNumHash, BlockNumber, Receipt, SealedBlockWithSenders, TxHash};
+use std::{collections::BTreeMap, fmt::Formatter, sync::Arc};
+use tokio::sync::broadcast::{Receiver, Sender};
+
+/// Trait that holds blocks and post state of execution those blocks.
+#[auto_impl(&, Arc)]
+pub trait SubChain: Send + Sync {
```

Comment:
We only have 1 implementation of this, do we expect different impls, besides `Chain`?

I think having this trait is beneficial in case the Chain model changes.
and using Arc<dyn> isn't a big deal

Reply:
I wanted to abstract it, but in the end, I moved `Chain` inside provider.